data/relationships.js

```diff
@@ -0,0 +1,114 @@
+ */
+ async function createRelationship(relationshipDescription, mentor, mentee, relationshipCategory){
+    //TODO: Validate inputs
+    validate.checkArgLength(arguments, 4);
+    validate.checkIsEmptyString(relationshipDescription); // Cannot be empty
+    mentor = validate.convertID(mentor); // TODO: decide if this should be a string or object id
+    mentee = validate.convertID(mentee);
+
+    // validate array content relationshipCategory
+
+    let relationshipDB = await relationshipCollection();
+
+    let workspaceId = null;
+    let chatChannel = null;
+
+    let createdOn = new Date(); // a number representing the milliseconds elapsed since epoch
+    let updatedOn = createdOn; // Initially will be the same as createdOn
+
+    const relationshipObject = {
+        _id: null,
+        relationshipDescription: relationshipDescription,
+        mentor: mentor,
+        mentee: mentee,
+        status: status.PENDING,
+        workspaceId: workspaceId,
+        createdOn: createdOn,
+        updatedOn: updatedOn,
+        relationshipCategory: relationshipCategory,
+        chatChannel: chatChannel
+    }
+
+    let insertedRelationship = await relationshipDB.insertOne(relationshipObject);
+    let relationshipId = await getRelationshipById(insertedRelationship.insertedId.toString());
+
+    return relationshipId
+ }
+
+ /**
+  * Retrieve a relationship object given an object ID
+  * @param {string} relationshipId An objectId associated with a relationship
+  * @returns {object} relationshipObject corresponding to the given id
+  */
+ async function getRelationshipById(relationshipId){
+    //Validating arguments
+    validate.checkArgLength(arguments, 1);
+    relationshipId = validate.convertID(relationshipId);
+
+    let relationshipDB = await relationshipCollection();
+    let foundRelationships = await relationshipDB.find({'_id': relationshipId}).toArray();
+    
+    if (foundRelationships.length === 0) throw `Error: no relationship with id '${relationshipId}' in the database`;
+    foundRelationships[0]._id = foundRelationships[0]._id.toString();
+
+    return foundRelationships[0]; // There should only be one relationship in the db with the given id
+ }
+
+
+/**
+ * Update the relationship status of a relationshipObject, after creation, the only thing about a relationship 
+ * that should be able to be changed is the status. That will be done on the user end by accepting a mentorship request,
+ * rejecting it, terminating it etc. 
+ * @param {string} relationshipId: An objectId associated with the relationship that needs to be changed
+ *  @param {status} newStatus: The changed status of the relationship
+ */
+ async function updateRelationshipStatus(relationshipId, newStatus){
+    // Validate arguments
+    validate.checkArgLength(arguments, 2);
+    relationshipId = validate.convertID(relationshipId);
+    if (!(newStatus instanceof status)) throw `Error: newStatus must be a status enum`;
```

Comment:
Does this mean we are going to convert the status value from "string" to "Enum" in Routes?
Because I believe data is received from the Front end in String format.